src/io/base.js

```diff
@@ -0,0 +1,47 @@
+import { singleLineString } from '../utils';
+
+/*
+ * Base class for io operations for both an Xpi or
+ * a directory
+ */
+export class IOBase {
+
+  getFile(path, streamOrString='string') {
+    switch (streamOrString) {
+      case 'stream':
+        return this.getFileAsStream(path);
+      case 'string':
+        return this.getFileAsString(path);
+      default:
+        throw new Error(singleLineString`Unexpected streamOrString
+          value "${streamOrString}" should be "string" or "stream"`);
+    }
+  }
+
+  getFilesByExt(...extensions) {
+
+    for (let ext of extensions) {
+      if (ext.indexOf('.') !== 0) {
+        // We use Promise.reject as we're not inside a `then()` or a
+        // Promise constructor callback.
+        // If we throw here it won't be caught.
+        return Promise.reject(new Error("File extension must start with '.'"));
+      }
+    }
+
+    return this.getFiles()
```

Comment:
In `BaseScanner` we define `_getContents()` (which other scanners must implement) as a simple function that throws a `NotImplentedError`.

Doing so here would be consistent and I just like outlining what the class should look like inside it without needing to look at a separate file.